lightning/src/ln/functional_tests.rs

```diff
@@ -1450,14 +1469,19 @@ fn do_channel_reserve_test(test_recv: bool) {
 		};
 
 		if test_recv {
-			let err = nodes[1].node.handle_update_add_htlc(&nodes[0].node.get_our_node_id(), &msg).err().unwrap();
-			match err {
-				LightningError{err, .. } => assert_eq!(err, "Remote HTLC add would put them over their reserve value"),
-			}
+			nodes[1].node.handle_update_add_htlc(&nodes[0].node.get_our_node_id(), &msg);
 			// If we send a garbage message, the channel should get closed, making the rest of this test case fail.
 			assert_eq!(nodes[1].node.list_channels().len(), 1);
 			assert_eq!(nodes[1].node.list_channels().len(), 1);
-			check_closed_broadcast!(nodes[1]);
+			let handle_error = &check_closed_broadcast!(nodes[1], true)[1];
+			if let &MessageSendEvent::HandleError { ref action, .. } = handle_error {
+				match action {
+					&ErrorAction::SendErrorMessage { ref msg } => {
+						assert_eq!(msg.data, "Remote HTLC add would put them over their reserve value");
```

Comment:
You can DRY this pattern up a ton (and fix a few cases where you forgot the trailing } else { panic!() }) by moving these matches into the macro.